nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/registry/flow/ImportAllFlows.java

```diff
@@ -0,0 +1,331 @@
+        listFlowVersions.initialize(context);
+    }
+
+    @Override
+    public String getDescription() {
+        return "From a provided directory as input, the directory content must be generated by the export-all-flows command, " +
+                "based on the file contents, the corresponding buckets, flows and flow versions will be created." +
+                "If not configured otherwise, already existing objects will be skipped.";
+    }
+
+    @Override
+    public StringResult doExecute(final NiFiRegistryClient client, final Properties properties) throws IOException, NiFiRegistryException, ParseException, CommandException {
+        final boolean skip = Boolean.parseBoolean(getRequiredArg(properties, CommandOption.SKIP_EXISTING));
+        final boolean isInteractive = getContext().isInteractive();
+
+        //Gather all buckets and create a map for easier search by bucket name
+        final Map<String, String> bucketMap = getBucketMap(client, isInteractive);
+
+        // Gather all flows and create a map for easier search by flow name.
+        // As flow name is only unique within the same bucket we need to use the bucket id in the key as well
+        final Map<Pair<String, String>, String> flowMap = getFlowMap(client, bucketMap, isInteractive);
+        final Map<Pair<String, String>, String> flowCreated = new HashMap<>();
+
+        // Gather all flow versions and create a map for easier search by flow id
+        final Map<String, List<Integer>> versionMap = getVersionMap(client, flowMap, isInteractive);
+
+        // Create file path list
+        final List<VersionFileMetaData> files = getFilePathList(properties);
+
+        // As we need to keep the version order the list needs to be sorted
+        files.sort((o1, o2) -> ComparisonChain.start()
+                .compare(o1.getBucketName(), o2.getBucketName())
+                .compare(o1.getFlowName(), o2.getFlowName())
+                .compare(o1.getVersion(), o2.getVersion())
+                .result());
+
+        for (VersionFileMetaData file : files) {
+            final String inputSource = file.getInputSource();
+            final String fileContent = getInputSourceContent(inputSource);
+            final VersionedFlowSnapshot snapshot = MAPPER.readValue(fileContent, VersionedFlowSnapshot.class);
+
+            final String bucketName = snapshot.getBucket().getName();
+            final String bucketDescription = snapshot.getBucket().getDescription();
+            final String flowName = snapshot.getFlow().getName();
+            final String flowDescription = snapshot.getFlow().getDescription();
+            final int flowVersion = snapshot.getSnapshotMetadata().getVersion();
+            // The original bucket and flow ids must be kept otherwise NiFi won't be able to synchronize with the NiFi Registry
+            final String flowId = snapshot.getFlow().getIdentifier();
+            final String bucketId = snapshot.getBucket().getIdentifier();
```

Comment:
Could we print a message at the beginning and end of this loop like?
```
Importing <flow Name> - <flowVersion> to <bucketName>
...
Successfully imported <flow Name> - <flowVersion> to <bucketName>
```
I think this would help keep track of the current messages to know where it is during the import.
I guess the successful message would only print if not skipping.